client/components/select/select.directive.js

```diff
@@ -0,0 +1,240 @@
+      return projectVersions.sort(projectVersionChoices);
+    }
+  })
+}).controller('UserSelectCtrl', function(api, $scope) {
+  // Always disable the create new feature on user select
+  $scope.createNew = false;
+
+  setupController($scope, api, {
+    defaults: {
+      modelProperty: $scope.multiple ? 'userIds' : 'userId',
+      label: 'Filter by user',
+      placeholder: 'All users',
+      extract: 'id'
+    },
+    fetchUrl: '/users'
+  });
+});
+
+/**
+ * Define the default attributes that the *-select directive have at least
+ */
+var defaultsAttributes = {
+  organization: '=',  // The organization for data scoping
+  modelObject: '=',   // The model object relates to an object where the selected values will be stored
+  modelProperty: '@', // Property name on the model object to store the selected values
+  prefix: '@',        // Prefix to created unique HTML IDs for labels and select fields
+  createNew: '=?',    // Enable the possibility to create a new item
+  autoSelect: '=?',   // Do an automatic selection on the first item when no other value is provided through the model object
+  placeholder: '@',   // Placeholder text shown in the select field
+  label: '@',         // The label text associated with the select field
+  noLabel: '=?',      // When set to true, the label is not shown
+  multiple: '=?',     // Switch between single and multiple select field
+  extract: '@',       // Name of the property to extract and use in the select field (Ex: id)
+  allowClear: '=?',   // Enable the possibility to remove a selection
+  labelNew: '@',      // The label text for the creation of a new item
+  defaultNewItem: '@' // The default text value for the creation of a new item
+};
+
+/**
+ * Define the standard default values for the select directive
+ */
+var defaults = {
+  showProperty: 'name', // Attribute name to use to show the item in the select field
+  allowClear: true,     // It is allowed to unselect items
+  multiple: false,      // She select fields are single selection
+  noLabel: false,       // The label is visible
+  extract: 'id'         // Use the ID to identify items
+};
+
+/**
+ * Create standard select directive
+ *
+ * @param controllerName The controller name to manage the directive
+ * @param attributes The attributes to configure the directive
+ * @returns The directive
+ */
+function createSelectDirective(controllerName, attributes) {
+  return {
+    restrict: 'E',
+    controller: controllerName,
+    templateUrl: '/templates/components/select/select.template.html',
+    scope: attributes
+  };
+}
+
+/**
+ * Configure a controller for a select directive
+ *
+ * @param $scope The scope
+ * @param api The API service
+ * @param options The options for the controller configuration
+ */
+function setupController($scope, api, options) {
+  if (!$scope.prefix) {
+    throw new Error("The prefix attribute on select directive is not set.");
+  }
+
+  if (!options) {
```

Comment:
Personally when I have sequences of validations like this, I use `else if` and don't put new lines in between. It's not interesting or complicated code so I don't want it taking space when the implementation below is more interesting.